From 85af0aee646d2200c0d513858fd64cfbc15f268f Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Wed, 7 Dec 2016 14:22:23 +0100
Subject: [PATCH 525/595] Bug 775691 - Paste eats new line marker

Don't use a workaround in the e_editor_dom_insert_html function if
inserting content that's coming from composer.
---
 .../web-extension/e-editor-dom-functions.c            | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/modules/webkit-editor/web-extension/e-editor-dom-functions.c b/modules/webkit-editor/web-extension/e-editor-dom-functions.c
index c8b7e90b82..f625e96517 100644
--- a/modules/webkit-editor/web-extension/e-editor-dom-functions.c
+++ b/modules/webkit-editor/web-extension/e-editor-dom-functions.c
@@ -8756,16 +8756,18 @@ e_editor_dom_insert_html (EEditorPage *editor_page,
 			 * empty after we insert the content. */
 			selection_marker = webkit_dom_document_get_element_by_id (
 				document, "-x-evo-selection-start-marker");
-			if (!webkit_dom_node_get_previous_sibling (WEBKIT_DOM_NODE (selection_marker))) {
-				WebKitDOMNode *sibling;
 
-				sibling = webkit_dom_node_get_next_sibling (WEBKIT_DOM_NODE (selection_marker));
-				sibling = webkit_dom_node_get_next_sibling (sibling);
-				if (WEBKIT_DOM_IS_HTML_BR_ELEMENT (sibling)) {
-					block = e_editor_dom_get_parent_block_node_from_child (WEBKIT_DOM_NODE (selection_marker));
-					remove_node (sibling);
+			if (!e_editor_page_is_pasting_content_from_itself (editor_page)) {
+				if (!webkit_dom_node_get_previous_sibling (WEBKIT_DOM_NODE (selection_marker))) {
+					WebKitDOMNode *sibling;
+
+					sibling = webkit_dom_node_get_next_sibling (WEBKIT_DOM_NODE (selection_marker));
+					sibling = webkit_dom_node_get_next_sibling (sibling);
+					if (WEBKIT_DOM_IS_HTML_BR_ELEMENT (sibling))
+						remove_node (sibling);
 				}
 			}
+			block = e_editor_dom_get_parent_block_node_from_child (WEBKIT_DOM_NODE (selection_marker));
 
 			e_editor_dom_selection_restore (editor_page);
 		}
-- 
2.15.0

